Userland/Libraries/LibCore/EventLoop.cpp

```diff
@@ -302,7 +302,8 @@ class InspectorServerConnection : public Object {
 };
 
 EventLoop::EventLoop([[maybe_unused]] MakeInspectable make_inspectable)
-    : m_private(make<Private>())
+    : m_wake_pipe_fds(&s_wake_pipe_fds)
```

Comment:
What APIs need to access the thread local wake pipes without going through an event loop? Signals? It would be nice to be able to not have every event loop in the thread use the same wake pipes, but... Hmm. Wondering if having each event loop own an actual wake pipe instead of holding a reference to the thread local ones would be better? Then the event loop could just add both wake pipes to the select list. The thread local ones could be reserved for signal notifications (unlikely) and cross thread or callback based wake notifications would use the member variable?

Reply:
Mostly worried about nested event loops like when GUI apps open a message box accidentally waking the "wrong" event loop

Reply:
Wait what am I talking about there's an event loop stack right? Maybe I'm just taking nonsense 🤔

Reply:
Two event loops in one thread can't run at the same time. Due to the stacking, it would be a lot of extra effort for nothing for each event loop to have its own wake pipe. The useful thing about thread-local wake pipes is that a newly stacked event loop will just be receiving events that are "targeted" at lower event loops which don't run right now. That's how it was previously and we now only extend the concept to multiple threads, ensuring that any event loop can receive wakes from anywhere.

The GUI apps also don't call the wake anyways, they just call pump_event or similar, which goes to the current thread's topmost event loop by default. So that behavior doesn't change. The actual use case that this commit fixes isn't part of this PR but of my other WIP branch.